src/main/java/org/apache/commons/compress/utils/BoundedNIOInputStream.java

```diff
@@ -0,0 +1,97 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package org.apache.commons.compress.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+/**
+ * NIO backed bounded input stream for reading a predefined amount of data from.
+ * @since 1.21
+ */
+public abstract class BoundedNIOInputStream extends InputStream {
```

Comment:
I agree that the name of the class it not perfect. Just wan't to share the reason why I took the name.

The newly defined read method reads the content into an `java.nio.ByteBuffer` and the `read` methods of the `InputStream` are calling it. So the name comes more from the fact that an inherited class only needs to implement the new `read` method to the `ByteBuffer` and not from the actual underlying implementation. My experience shows that reading into a ByteBuffer from a non NIO source is not so much fun and therefore I put the `NIO` in the classname.

If someone has a better suggestion for a classname it would be welcome.

Reply:
Thanks for explaining your rationale @theobisproject . I don't have a better suggestion, but I agree with @PeterAlfredLee that it is confusing. Perhaps the old name?

Reply:
Since the `BoundedInputStream` class is still there how about naming it `BoundedArchiveInputStream`?

Reply:
+1
I don't have a better idea.